libsync/SyncMaster.cpp

```diff
@@ -534,10 +546,10 @@ void SyncMaster::maintainPeersConnection()
     });
 
     // If myself is not in group, ignore receive packet checking from all peers
-    if (memberSet.find(m_nodeId) == memberSet.end())
-        m_msgEngine->needCheckPacketInGroup = false;
-    else
-        m_msgEngine->needCheckPacketInGroup = true;
+    bool hasMyself = false;
+    for (auto const& mo : minerOrObserver)
```

Comment:
do you know there is a function named exist defined in map/set?

Reply:
Sorry, I thought the minerOrObserver is a map or set.
What I think is that you don't need to traverse the list again, may be you can figure it out when you traversed last time

Reply:
Use find instead. About map/set, to detected existing, maybe also use find() == end()

Reply:
Sure, fix it immediately